subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
+
+	if (!iface) {
+		NET_ERR("Cannot bind to %s",
+			net_sprint_ipv6_addr(&addr6->sin6_addr));
+
+		return -EADDRNOTAVAIL;
+	}
+
+	if (dest_addr->family == AF_INET6) {
+		if (net_sin6_ptr(&context->local)->sin6_addr) {
+			return 0;
+		}
+
+		net_context_set_iface(context, iface);
+		net_sin6_ptr(&context->local)->sin6_family = AF_INET6;
+		net_sin6_ptr(&context->local)->sin6_addr =
+			(struct in6_addr *)net_ipv6_unspecified_address();
+
```

Comment:
We need to check the port number here too. The net_context_get() sets the initial value for the port, but it might be used already when we get into this point. So we need to check it by calling check_used_port() like the net_context_bind() is doing, and probably call find_available_port() again if the port was used already (unlikely to happen thou).